src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4498,7 +4492,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           case _                              => false
         }
       }
-      acceptsApplyDynamicWithType(qual, name) map { tp =>
+      acceptsApplyDynamicWithType(qual, name).map { _ =>
         // If tp == NoType, pass only explicit type arguments to applyXXX.  Not used at all
```

Comment:
`tp` is mentioned in the obscure comment

Reply:
Huh, we have special accommodation for scala-virtualized? It's kinda dead.
Also how could all of this be used when it's private?
I think `acceptsApplyDynamicWithType` should just return a boolean as the name implies.

Reply:
Also this is an instance of "the invisible gorilla"